crypto/property/property.c

```diff
@@ -67,6 +72,17 @@ DEFINE_SPARSE_ARRAY_OF(ALGORITHM);
 static void ossl_method_cache_flush(OSSL_METHOD_STORE *store, int nid);
 static void ossl_method_cache_flush_all(OSSL_METHOD_STORE *c);
 
+static int ossl_method_up_ref(METHOD *method)
+{
+    return method->up_ref != NULL ? (*method->up_ref)(method->method) : 1;
+}
+
+static void ossl_method_free(METHOD *method)
+{
+    if (method->free != NULL)
+        (*method->free)(method->method);
+}
```

Comment:
Re both up_ref and free, I think you shouldn't check for NULL, 'cause internal.  If your code hasn't set things up correctly, it deserves a _KABOOM_.  That may also help you find intricate stoopid bugs quicker.

Reply:
I was thinking of making these call mandatory through out -- i.e. the pointers can never be NULL.

Taking out the null checks isn't helping.

Reply:
I'm guessing but I've a feeling the store needs to clean the cache better -- without ref counting in the cache, it won't matter but with counting it does.

Something for when I'm feeling more alert.